### PR TITLE
Fixes #75 - Cleanup stale file handle

### LINKED ISSUE #75: worker.GatherResults doesn't make it possible to close the results file
It calls DoRequest and returns an `io.Reader`, so it never closes the file it opens.

### PR DESCRIPTION
**Pull request by timothysc**
Fixes #75 
Signed-off-by: Timothy St. Clair <timothysc@gmail.com>

### CODE REVIEW

File: pkg/worker/worker.go [hunk 1]
```diff
@@ -36,6 +36,7 @@ import (
 func GatherResults(waitfile string, url string) error {
 	var inputFileName []byte
 	var err error
+	var outfile * os.File
```
Comment (chuckha) [inline pkg/worker/worker.go]: gofmt this bad boy

Comment (timothysc) [reply pkg/worker/worker.go]: It does, on save.. 

    "go.formatOnSave": true, 
    "go.formatTool": "gofmt",

File: pkg/worker/worker.go [hunk 2]
```diff
@@ -61,8 +62,14 @@ func GatherResults(waitfile string, url string) error {
 	}
 
 	// transmit back the results file.
-	return DoRequest(url, func() (io.Reader, error) {
-		outfile, err := os.Open(s)
-		return outfile, errors.WithStack(err)
+	err = DoRequest(url, func() (io.Reader, error) {
+		outfile, e2 := os.Open(s)
+		return outfile, errors.WithStack(e2)
```
Comment (davecheney) [inline pkg/worker/worker.go]: I'm not sure about this; outfile on line 66 is in a different scope to line 39, so on line 70, outfile will always be nil.

I suggest

s/outfile/f here to make it clear that you're talking about a different variable.

File: pkg/worker/worker.go [hunk 3]
```diff
@@ -36,6 +36,7 @@ import (
 func GatherResults(waitfile string, url string) error {
 	var inputFileName []byte
 	var err error
```
Comment (davecheney) [inline pkg/worker/worker.go]: delete this line, use err := on line 65

File: pkg/worker/worker.go (hunk 1 above)
Comment (chuckha) [reply pkg/worker/worker.go]: weird!! When I save it it moves that * over to be `var outfile *os.File`

I literally don't care on this issue. Feel free to push.

**Comment by timothysc** [comment]
Thx for the quick review.

File: pkg/worker/worker.go [hunk 4]
```diff
@@ -36,6 +36,7 @@ import (
...
+	var outfile *os.File
 
 	// just loop looking for a file.
 	done := false
```
Comment (davecheney) [inline pkg/worker/worker.go]: I know this is out of scope, but what about

for {
 // do work
 if err == nil {
     // success!!
     break
 }
 // log and wait
}

File: pkg/worker/worker.go [hunk 5]
```diff
@@ -60,9 +61,16 @@ func GatherResults(waitfile string, url string) error {
 		url += "." + filenameParts[1]
```
Comment (davecheney) [inline pkg/worker/worker.go]: line 45: inputFileName is misleading, it's not the name of a file, it's the contents of waitfile.

File: pkg/worker/worker.go (hunk 4 above)
Comment (timothysc) [reply pkg/worker/worker.go]: Patches welcome ;-)  
fwiw I didn't write this part of the code, I'm just trying to throw some fixes in before release.

**Comment by davecheney** [comment]
Noted, I'll be the change I seek.

**Pull request merged**